osu.Game/Overlays/News/NewsHeader.cs

```diff
@@ -3,44 +3,46 @@
 
 using osu.Framework.Bindables;
 using osu.Framework.Graphics;
-using System;
 
 namespace osu.Game.Overlays.News
 {
     public class NewsHeader : BreadcrumbControlOverlayHeader
     {
-        private const string front_page_string = "frontpage";
+        public const string FRONT_PAGE_STRING = "frontpage";
 
-        public readonly Bindable<string> Post = new Bindable<string>(null);
-
-        public Action ShowFrontPage;
+        public readonly Bindable<string> Post = new Bindable<string>(FRONT_PAGE_STRING);
 
         public NewsHeader()
         {
-            TabControl.AddItem(front_page_string);
+            TabControl.AddItem(FRONT_PAGE_STRING);
+            Current.Value = FRONT_PAGE_STRING;
+            Current.BindValueChanged(onCurrentChanged);
+            Post.BindValueChanged(onPostChanged, true);
+        }
 
-            Current.ValueChanged += e =>
-            {
-                if (e.NewValue == front_page_string)
-                    ShowFrontPage?.Invoke();
-            };
+        public void SetFrontPage() => Post.Value = FRONT_PAGE_STRING;
 
-            Post.ValueChanged += showPost;
+        public void SetArticle(string slug) => Post.Value = slug;
+
+        private void onCurrentChanged(ValueChangedEvent<string> current)
+        {
+            if (current.NewValue == FRONT_PAGE_STRING)
+                Post.Value = FRONT_PAGE_STRING;
         }
 
-        private void showPost(ValueChangedEvent<string> e)
+        private void onPostChanged(ValueChangedEvent<string> post)
         {
-            if (e.OldValue != null)
-                TabControl.RemoveItem(e.OldValue);
+            if (post.OldValue != FRONT_PAGE_STRING)
+                TabControl.RemoveItem(post.OldValue);
 
-            if (e.NewValue != null)
+            if (post.NewValue != FRONT_PAGE_STRING)
             {
-                TabControl.AddItem(e.NewValue);
-                Current.Value = e.NewValue;
+                TabControl.AddItem(post.NewValue);
+                Current.Value = post.NewValue;
             }
             else
             {
-                Current.Value = front_page_string;
+                Current.Value = FRONT_PAGE_STRING;
             }
```

Comment:
breadcrumb code should not be relying on a magic string to tell whether or not it should be going back to the front page. `null` was actually better than what is here currently but both ways are definitively worse than, again, not having a public bindable in which one magic value represents the front page, and instead actually having proper separate methods to show the front page and to show a single post.